src/rail_gui.cpp

```diff
@@ -772,6 +772,36 @@ struct BuildRailToolbarWindow : Window {
 	static HotkeyList hotkeys;
 };
 
+/**
+ * Selects RailType for autorail based on _last_built_railtype and global modifiers from RailToolbarWidgetModifiers
+ *
+ * Without wrapping, so if you're on RAILTYPE_RAIL you'll stay there. Same for upper boundary.
+ * Also clears modifier bits to allow normal widget OhHotkey logic.
+ *
+ * @param hotkey Hotkey
+ * @return selected RailType
+ */
+static RailType ChooseRailTypeOnGlobalHotkey(int &hotkey)
+{
+	extern RailType _last_built_railtype;
+	RailType rt = _last_built_railtype;
+	if (HasBit(hotkey, GHK_MOD_RAT_PREV_RAILTYPE))
+	{
+		ClrBit(hotkey, GHK_MOD_RAT_PREV_RAILTYPE);
+		do { rt--; } while (rt > RAILTYPE_BEGIN && !HasRailtypeAvail(_local_company, rt));
```

Comment:
I get why you've done it this way, but I think I'd prefer this (and the other) to be styled "properly"
```suggestion
		do {
			rt--;
		} while (rt > RAILTYPE_BEGIN && !HasRailtypeAvail(_local_company, rt));
```